lib/engine/game/g_1848/step/buy_train.rb

```diff
@@ -17,14 +31,23 @@ def can_entity_buy_train?(entity)
           def buyable_trains(entity)
             # Cannot buy 2E if one is already owned
             owns_2e = entity.trains.any? { |t| t.name == '2E' }
-            return super unless owns_2e
+            return super if !owns_2e && @game.phase.status.include?('can_buy_trains')
```

Comment:
Going to need help with this method.
I'm combining two logical conditions:
1- include the 2e only if player already doesn't own the 2e
2- include only depot trains if phase 3 (buy from other corps) hasn't triggered.
This code works, but it's ugly and hard to understand